apps/condo/domains/organization/components/EmployeeRolesSettingsContent.tsx

```diff
@@ -1,62 +1,629 @@
+import { Close } from '@open-condo/icons'
 import { useIntl } from '@open-condo/next/intl'
 import { useOrganization } from '@open-condo/next/organization'
+import { ActionBar, ActionBarProps, Button, Checkbox, Tooltip } from '@open-condo/ui'
 
-import { DEFAULT_PAGE_SIZE, Table, TableRecord } from '@condo/domains/common/components/Table/Index'
-import { Tooltip } from '@condo/domains/common/components/Tooltip'
-import { getPageIndexFromOffset, parseQuery } from '@condo/domains/common/utils/tables.utils'
-import { useEmployeeRolesTableColumns } from '@condo/domains/organization/hooks/useEmployeeRolesTableColumns'
+import {
+    EXPANDABLE_COLUMN_STUB,
+    ExpandableTable,
+    Table,
+    TableRecord,
+} from '@condo/domains/common/components/Table/Index'
+import { B2BAppContext, B2BAppPermission, B2BAppRole } from '@condo/domains/miniapp/utils/clientSchema'
+import {
+    GROUP_NAME_COLUMN_WIDTH,
+    ROLE_COLUMN_STYLE,
+    useEmployeeRolesTableColumns,
+    useEmployeeRoleTicketVisibilityInfoTableColumns,
+} from '@condo/domains/organization/hooks/useEmployeeRolesTableColumns'
 import { OrganizationEmployeeRole } from '@condo/domains/organization/utils/clientSchema'
 
-const StyledTable = styled(Table)`
-  .ant-table-cell-ellipsis {
-    white-space: inherit;
-  }
-  
-  .ant-table-cell-row-hover {
-    background-color: inherit !important;
-    cursor: not-allowed;
-  }
-`
 
-const MEDIUM_VERTICAL_GUTTER: [Gutter, Gutter] = [0, 40]
+const MEDIUM_VERTICAL_GUTTER: RowProps['gutter'] = [0, 40]
 
-export const EmployeeRolesSettingsContent = () => {
+type PermissionRow = {
+    id: string,
+    key: string,
+    name: string,
+}
+
+type PermissionsGroup = {
+    id: string,
+    groupName: string
+    permissions: PermissionRow[],
+}
+
+type EmployeeRolesTableProps = {
+    connectedB2BApps: B2BApp[],
+    employeeRoles: OrganizationEmployeeRoleType[],
+    b2BAppRoles: B2BAppRoleType[],
+    b2BAppPermissions: B2BAppPermissionType[],
+    loading: boolean,
+    createB2BAppRoleAction: IUseCreateActionType<B2BAppRoleType, B2BAppRoleCreateInput>,
+    softDeleteB2BAppRoleAction: IUseSoftDeleteActionType<B2BAppRoleType>,
+    updateB2BAppRoleAction: IUseUpdateActionType<B2BAppRoleType, B2BAppRoleUpdateInput>,
+}
+
+type PermissionsType = { [permissionKey: string]: boolean }
+
+type B2BAppPermissionsState = {
+    roleId?: string,
+    permissions: PermissionsType
+}
+
+type PermissionsState = {
+    [roleId: string]: {
+        organizationPermissions: PermissionsType,
+        b2bAppRoles: {
+            [b2bAppId: string]: B2BAppPermissionsState
+        }
+    }
+}
+
+type TableCheckboxProps = {
+    employeeRoleId: string
+    b2bAppId?: string
+    permissionKey: string
+    permissionsState: PermissionsState
+    setPermissionsState: Dispatch<SetStateAction<PermissionsState>>
+}
+
+/**
+ * Changes value of the checkbox and related checkboxes.
+ * If user set "canManage" permission checkbox, then "canRead" checkbox sets to true automatically.
```

Comment:
Does this logic applied to miniapps?

Seems like it shouldn't. Because in condo we have strong naming agreement. Our contractors don't have one. So it can possibly be any key, not just `canRead` / `canManage`.

Reply:
Also aren't there any warnings? Seems like function has `Record<string, any>: any` type

Reply:
> Does this logic applied to miniapps?
> 
> Seems like it shouldn't. Because in condo we have strong naming agreement. Our contractors don't have one. So it can possibly be any key, not just `canRead` / `canManage`.


`canManage` here it's just an example: if we have not checked `canRead` checkbox (for b2bApps it's only on client-side)  and set any other checkbox in the group for employee role, then `canRead` checkbox becomes true.

About name collisions: in the current logic I find it difficult to represent a case when we have `b2bAppRole` (= can read app) and it has a separate right `canRead{AppId}` = false. Contractors have created such right and it means something other than the ability to view app? So it’s bad naming of this right. And if, logically, this right should mean the same as here, then it should not exist at all, as we are responsible for this having `B2bAppRole` (And even if there is nothing will break).

> Also aren't there any warnings? Seems like function has `Record<string, any>: any` type

Added function return type `PermissionsType`.